src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,90 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDos;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+
+/**
+ * Deals with making sense of the user command.
+ */
+
+public class Parser {
+    TaskList taskList;
+    Ui ui;
+    String line;
+    public Parser(TaskList taskList, Ui ui) {
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    /**
+     * Deals with processing the user command and taking actions accordingly.
+     *
+     * @param line line being processed.
+     * @throws DukeException  If input is incorrect.
+     * @throws ParseException  If date is not in date format.
+     */
+
+    public void process(String line) throws DukeException, ParseException {
+        Integer num; //number in list which is done
+        Task currTask; //refers to current task in list
+        String currEvent;
+        String desc = ""; //current task being added
+        String time = ""; //current time of current task being added
+        String[] words = line.split(" ", 2);
+        currEvent = words[0];
+        if (line.equals("list")) {
+            ui.printList(taskList.list);
+        } else if (words[0].equals("done")) {
+            num = Integer.valueOf(words[1]);
+            currTask = taskList.list.get(num - 1);
+            currTask.setStatusIcon(true);
+            ui.printDone(currTask);
```

Comment:
Might want to consider creating a Command class and have the logic carried out somewhere else so that this Parser class is solely responsible for parsing user inputs.